proxy/errors.go

```diff
@@ -0,0 +1,33 @@
+package proxy
+
+import (
+	"fmt"
+	"net/http"
+)
+
+type OpenFaaSError struct {
+	StatusCode int
+	Message    string
```

Comment:
I think it would be good to include a copy of the response headers here as well, depending on the configuration of the system there could be useful debugging/tracing headers or other data that the caller would want to inspect when there is an error.  In that case it is probably a good idea to change 

```go
func NewOpenFaaSError(message string, statusCode int) error {}
```

to

```go
func NewOpenFaaSError(message string, resp *http.Response) error {}
```